feat: added JMeter docs (#2654)
kubeshop/testkube: docs/docs/4-test-types/executor-jmeter.md

---
sidebar_position: 4
sidebar_label: JMeter
---
# JMeter Performance Tests

The Apache JMeter™ application is open source software, a 100% pure Java application designed to load test functional behavior and measure performance. It was originally designed for testing Web Applications but has since expanded to other test functions.

## What can I do with it?

> Apache JMeter may be used to test performance both on static and dynamic resources, Web dynamic applications.

Reviewer: ```suggestion
Apache JMeter may be used to test performance both on static and dynamic resources and Web dynamic applications.
```

          <boolProp name="HTTPSampler.auto_redirects">false</boolProp>
          <boolProp name="HTTPSampler.use_keepalive">true</boolProp>
          <boolProp name="HTTPSampler.DO_MULTIPART_POST">false</boolProp>
          <stringProp name="HTTPSampler.embedded_url_re"></stringProp>
          <stringProp name="HTTPSampler.connect_timeout"></stringProp>
          <stringProp name="HTTPSampler.response_timeout"></stringProp>
        </HTTPSamplerProxy>
        <hashTree>
          <ResponseAssertion guiclass="AssertionGui" testclass="ResponseAssertion" testname="Response Assertion" enabled="true">
            <collectionProp name="Asserion.test_strings">
              <stringProp name="-1081444641">Testkube</stringProp>
            </collectionProp>
            <stringProp name="Assertion.custom_message"></stringProp>
            <stringProp name="Assertion.test_field">Assertion.response_data</stringProp>
            <boolProp name="Assertion.assume_success">false</boolProp>
            <intProp name="Assertion.test_type">16</intProp>
          </ResponseAssertion>
          <hashTree/>
        </hashTree>
      </hashTree>
    </hashTree>
  </hashTree>
</jmeterTestPlan>

```

> Testkube JMeter executor accept a test file as an input.

Reviewer: ```suggestion
The Testkube JMeter executor accepts a test file as an input.
```

```bash
kubectl testkube create test --file test.jmx --name jmeter-test --type jmeter/test
```
You don't need to pass a type here, Testkube will autodetect it. 

> To run the test, pass previously created test name: 

Reviewer: ```suggestion
To run the test, pass the previously created test name: 
```

```bash 
kubectl testkube run test -f jmeter-test
```

> You can also create a Test based on Git repository:

Reviewer: ```suggestion
You can also create a Test based on a Git repository:
```

```bash
# create test from this Git repository
kubectl testkube create test --git-uri https://github.com/kubeshop/testkube-executor-jmeter.git --git-branch main --git-path examples/kubeshop.jmx --type jmeter/test --name jmeter-test-from-git
```

Testkube will clone the repository and create a Testkube Test Custom Resource in your cluster automatically on each test run. 

### **Using Additional JMeter Arguments in Your Tests**

> You can also pass additional arguments to `jmeter` binary thanks to `--args` flag:

Reviewer: ```suggestion
You can also pass additional arguments to the `jmeter` binary thanks to the `--args` flag:
```

```bash
$ kubectl testkube run test -f jmeter-test --args '-LsutHost=https://staging.kubeshop.com -LsomeParam=someValue'
```

### **JMeter Test Results**

A JMeter test will be successful in Testkube when all checks and thresholds are successful. In the case of an error, the test will have `failed` status,

> JMeter executor is configured to store `report.jtl` file after test run, you can get it from "Artifacts" tab in execution results in Testkube UI, 

Reviewer: ```suggestion
JMeter executor is configured to store the `report.jtl` file after the test run. You can get the file from the "Artifacts" tab in the execution results in Testkube UI, 
```

> or download it with `testkube get artifacts EXECUTION_ID` command.

Reviewer: ```suggestion
or download it with the `testkube get artifacts EXECUTION_ID` command.
```
